frontend/components/instruments/SpatialInstrumentInternal.js

```diff
@@ -0,0 +1,180 @@
+import React, {useState, useRef, useEffect} from "react";
+import PropTypes from "prop-types";
+import SoundPoint from "./SoundPoint";
+import SamplePlayer from "./SamplePlayer";
+import STYLES from "./SpatialInstrument.module.scss";
+
+/* TODO: there is a bug in which the player crashes
```

Comment:
This is an underlying problem in `SamplePlayer`: it should probably reload the audio buffer at the end of playback, regardless of whether it's going to play again, and we should probably cache the decoded buffer rather than decoding every time we need to load it. I'll open an issue: not something we're going to fix this week, but important follow-up work to make all of the audio playback here robust.